plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1186,6 +1186,15 @@ public void testSchemaEvolution()
         dropTable("test_schema_evolution_drop_middle");
     }
 
+    @Test
+    public void testDropPartitionColumn()
+    {
+        String tableName = "test_drop_partition_column_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col0 INTEGER, col1 INTEGER, col2 INTEGER) WITH (partitioning = Array['col2'])");
+        assertQueryFails("ALTER TABLE " + tableName + " DROP COLUMN col2", "Failed to drop column: Can not drop partition field:.*");
```

Comment:
Dropping partitioned columns was disallowed in Iceberg library side even before this change, right?

Reply:
Yes, It was disallowed. But the message which was thrown was `"Cannot find source column for partition field: *"`.

Reply:
Could you change the commit title as "Improve error message when trying to drop partitioned columns in Iceberg" or something?

Reply:
I confirmed Iceberg allowed dropping partitioned columns if the transform is `void` and the spec says:
> The void transform may be used to replace the transform in an existing partition field so that the field is effectively dropped in v1 tables.

@alexjo2144 @krvikash Do you agree to keep the above behavior?

Reply:
I'm not sure if this ever got fixed: https://github.com/apache/iceberg/issues/4563
The issue was closed for inactivity.
But if it didn't, dropping a column like that may corrupt your table.

Reply:
> The void transform may be used to replace the transform in an existing partition field so that the field is effectively dropped in v1 tables.

Hi @ebyhr, I think we already have this behavior We can drop a partition column in trino that is part of the older partition spec. This refers to https://github.com/apache/iceberg/issues/4563

```sql
CREATE TABLE test2 (id INTEGER, name VARCHAR) WITH (partitioning = ARRAY['id', 'truncate(name, 5)']);
ALTER TABLE test2 SET PROPERTIES partitioning = ARRAY['id'];
ALTER TABLE test2 DROP COLUMN name;
```

The current PR refers to the issue when we are trying to drop a partition column that is part of the current partition spec.

Spark SQL uses `ALTER TABLE ... DROP PARTITION FIELD ...` statement to remove the column from partition spec, whereas trino uses `ALTER TABLE ... SET PROPERTIES partitioning = ...`.

Reply:
However, Fails to insert into iceberg table after dropping a partition column that was used by older partition specs. Raised #15729

Reply:
Thanks, @krvikash for pointing it out. I have added `isVoidTransform` to check for void transform.